lib/rebaser-server/src/server.rs

```diff
@@ -179,27 +138,26 @@ impl Server {
             shutdown_watch_rx,
             external_shutdown_tx,
             graceful_shutdown_rx,
-            rabbitmq_config,
+            messaging_config,
             content_store_pg_pool,
         })
     }
 
     /// The primary function for running the server. This should be called when deciding to run
     /// the server as a task, in a standalone binary, etc.
     pub async fn run(self) -> ServerResult<()> {
-        management_loop::management_loop_infallible_wrapper(
-            self.recreate_management_stream,
+        core_loop::setup_and_run_core_loop(
             self.pg_pool,
             self.nats,
             self.veritech,
             self.job_processor,
             self.symmetric_crypto_service,
             self.encryption_key,
             self.shutdown_watch_rx,
-            self.rabbitmq_config,
+            self.messaging_config,
             self.content_store_pg_pool,
         )
-        .await;
+        .await?;
```

Comment:
It's only fallible in the setup phase. The loop itself is infallible within our control.